core/src/main/java/io/ddf/analytics/ABinningHandler.java

```diff
@@ -22,21 +30,192 @@ public ABinningHandler(DDF theDDF) {
   public abstract List<AStatisticsSupporter.HistogramBin> getVectorApproxHistogram(String column, int numBins)
           throws DDFException;
 
-  public DDF binning(String column, String binningType, int numBins, double[] breaks, boolean includeLowest,
-      boolean right) throws DDFException {
+  public DDF binning(String column, String binningType, int numBins, double[] breaks, boolean toLabels, boolean includeLowest,
+      boolean right, int precision) throws DDFException {
+
+    Schema.Column colMeta = this.getDDF().getColumn(column);
+
+    BinningType binType = BinningType.get(binningType);
+
+    double[] theBreaks = breaks;
+
+    switch(binType) {
+      case CUSTOM:
+        if (breaks == null) throw new DDFException("Please enter valid break points");
+        // check for monotonic
+        Arrays.sort(theBreaks);
+        if (!Arrays.equals(theBreaks, breaks)) throw new DDFException("Please enter increasing breaks");
```

Comment:
Does this work? `Arrays.sort()` is in-place, and `theBreaks` is just a reference to `breaks`, so after this `Arrays.equals(theBreaks, breaks)` will be true anyway?
You might need to clone `breaks` somehow.